### PR TITLE
Fix duplicate test runs when using the `skip-random-tests` and `parallel-tests` profiles

### LINKED ISSUE #8637: Tests are run multiple times when using profiles `skip-random-tests` and `parallel-tests`
**Description**

When using the `skip-random-tests` profile in combination with the `parallel-tests` profile, some tests are run multiple times. 
To verify, we can run the engine tests and filter the output for duplicate lines:
```
$ mvn verify -pl engine -Pskip-random-tests,parallel-tests | sort | uniq -d
[INFO] Running io.camunda.zeebe.engine.processing.bpmn.activity.UserTaskTransformerTest
[INFO] Running io.camunda.zeebe.engine.processing.bpmn.activity.UserTaskTransformerTest$AssignmentDefinitionTests
[INFO] Running io.camunda.zeebe.engine.processing.bpmn.activity.UserTaskTransformerTest$AssignmentDefinitionTests$AssigneeTests
[INFO] Running io.camunda.zeebe.engine.processing.bpmn.activity.UserTaskTransformerTest$AssignmentDefinitionTests$CandidateGroupsTests
[INFO] Running io.camunda.zeebe.engine.processing.common.ExpressionProcessorTest
[INFO] Running io.camunda.zeebe.engine.processing.common.ExpressionProcessorTest$EvaluateArrayOfStringsExpressionTest
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_1.MessageSubscriptionSentTimeMigrationTest
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_1.MessageSubscriptionSentTimeMigrationTest$BlackboxTest
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_1.MessageSubscriptionSentTimeMigrationTest$MockBasedTests
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_1.ProcessMessageSubscriptionSentTimeMigrationTest
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_1.ProcessMessageSubscriptionSentTimeMigrationTest$BlackboxTest
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_1.ProcessMessageSubscriptionSentTimeMigrationTest$MockBasedTests
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_3.TemporaryVariableMigrationTest
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_3.TemporaryVariableMigrationTest$BlackboxTest
[INFO] Running io.camunda.zeebe.engine.state.migration.to_1_3.TemporaryVariableMigrationTest$MockBasedTests
```
Using the `skip-random-tests` or `parallel-tests` profiles on their own does not run tests multiple times. To verify, run
`mvn verify -pl engine -Pparallel-tests | sort | uniq -d` or `mvn verify -pl engine -Pskip-random-tests | sort | uniq -d` 

This might be the reason why sometimes the test report xml files are corrupted. In our Jenkins pipeline we are using both profiles to run the unit tests: https://github.com/camunda-cloud/zeebe/blob/ed46be39bc1dea97172a46e5f6b5791a222d0d17/.ci/scripts/distribution/test-java.sh#L29-L32

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

When adding custom excludes to the test discovery configuration of surefire, the default exclude for inner classes is no longer active. This caused issues as surefire will try to run nested test classes when it doesn't need to.

## Related issues

closes #8637

### CODE REVIEW

**Comment by npepinpe** [comment]
If I understand correctly, Surefire will not execute inner test classes? But don't we have some of these with the `@Nested` annotations? Are we not executing them? :scream: When I researched a little bit, I dug up a Junit 5 issue about this and the nested static classes, and they mention the workaround of simply putting an empty `<exclude/>`. Is this something we would also consider? Why do we need to add a pattern at all? From your pattern, aren't we excluding everything? I don't understand how these excludes work, I guess. Maybe let's have a short chat face to face, since I have a bunch of questions :smile:

**Comment by npepinpe** [comment]
I assume we'd probably want to backport this as well :+1:

File: parent/pom.xml
Comment (npepinpe) [inline parent/pom.xml:1415]: :upside_down_face: We typically add the comments above the line in POMs, mostly for readability

File: .ci/scripts/lib/duplicate-tests.sh
Comment (npepinpe) [inline .ci/scripts/lib/duplicate-tests.sh:7]: :wrench: I would propose to keep the same header as the `flaky-tests` script, e.g.

```bash
#######################################
# Parses Maven output to find if any tests was run more than once.
# Globals:
#   None
# Arguments:
#   1 - file containing the test outputs
#   2 - writable file which where the duplicate tests will be written
# Outputs:
#   Error message on STDERR if duplicate tests were found.
#   Error message on STDERR if no tests were found, indicating the function is broken
# Returns:
#   0 if no tests were run multiple times
#   1 if at least one test was run multiple times
#######################################
```

File: .ci/scripts/lib/duplicate-tests.sh
Comment (npepinpe) [inline .ci/scripts/lib/duplicate-tests.sh:16]:
```suggestion
          echo "Found duplicate test runs!">&2
```

:x: As this is an error message, we should print it out on STDERR.

File: .ci/scripts/lib/duplicate-tests.sh
Comment (npepinpe) [inline .ci/scripts/lib/duplicate-tests.sh:21]:
```suggestion
  echo "Could not find any tests, duplicate detection is broken.">&2
```

:x: As this is an error message, we should print it out on STDERR.

File: .ci/scripts/lib/duplicate-tests.sh
Comment (npepinpe) [inline .ci/scripts/lib/duplicate-tests.sh:22]: :upside_down_face:  Maybe we can consider outputting a different exit code? Though I guess it doesn't matter much as long as we print out the error message.

**Review by npepinpe — APPROVED** [review]
:rocket: 

Again, great catch! Only minor comments, don't think this would need another review, but please address them before merging.

- :+1: Good initiative/catch adding a lib script, when I don't think anyone ever told you about these, and they're easy to miss :smile:
- :x: Please save the `DuplicateTests.txt` as a build artifact in Jenkins on failure (a bit like how we save the error reports). This can be useful if there's ever any bug with the script, so we can diagnose issues. We didn't do that with flaky tests, but we have an issue to do that, so hopefully we learn from that mistake :smile:

File: .ci/scripts/lib/flaky-tests.sh
Comment (npepinpe) [inline .ci/scripts/lib/flaky-tests.sh:4]: I'm guessing this was unintentional? :smile:

File: Jenkinsfile
Comment (npepinpe) [inline Jenkinsfile:180]: Oh, thanks for adding the flaky tests as well!

File: .ci/scripts/lib/flaky-tests.sh
Comment (oleschoenburg) [reply .ci/scripts/lib/flaky-tests.sh:4]: :facepalm: yup, thanks! I still need to test the artifact archival anyway though.

File: Jenkinsfile
Comment (oleschoenburg) [reply Jenkinsfile:180]: Let's see if this works first :D

Comment (oleschoenburg) [reply Jenkinsfile:180]: Hm, if it's best practice I'll change it. Personally, I'm always annoyed when I have to unzip a single text file that's < 10KiB

Comment (npepinpe) [reply Jenkinsfile:180]: Leave it as is then, I can't remember the reasoning anymore - if it's an issue, we'll find out again :smile: It might've simply been that test reports, uncompressed, can be hundreds of megabytes.

File: .ci/scripts/lib/duplicate-tests.sh
Comment (oleschoenburg) [reply .ci/scripts/lib/duplicate-tests.sh:30]: Yeah I think the error message is enough.

**Comment by oleschoenburg** [comment]
Okay, I added one more fixup because i forgot to make this work for random tests and then everything should be good!

bors r+

**Comment by oleschoenburg** [comment]
:sob: I'll backport this manually then.

**Pull request merged**
